cpp/open3d/data/Downloader.cpp

```diff
@@ -102,135 +105,76 @@ static size_t WriteDataCb(void* ptr, size_t size, size_t nmemb, FILE* stream) {
     return written;
 }
 
-static std::string GetAbsoluteFilePath(const std::string& url,
-                                       const std::string& output_file_path,
-                                       const std::string& output_file_name) {
-    std::string file_name;
-    if (output_file_name.empty()) {
-        file_name = utility::filesystem::GetFileNameWithoutDirectory(url);
-    } else {
-        file_name = output_file_name;
+void DownloadFromURL(const std::string& url,
+                     const std::string& sha256,
+                     const std::string& prefix,
+                     const std::string& data_root) {
+    // Always print URL to inform the user. If the download fails, the user
+    // knows the URL.
+    utility::LogInfo("Downloading {}", url);
+
+    // Sanity checks.
+    if (sha256.size() != SHA256_DIGEST_LENGTH * 2) {
+        utility::LogError("Invalid sha256 length {}, expected to be {}.",
+                          sha256.size(), SHA256_DIGEST_LENGTH * 2);
     }
-
-    std::string file_path;
-    if (output_file_path.empty()) {
-        file_path = LocateDataRoot();
-    } else {
-        file_path = output_file_path;
+    if (prefix.empty()) {
+        utility::LogError("Download prefix cannot be empty.");
     }
 
-    // It will create the directory hierarchy if not present.
-    if (!utility::filesystem::DirectoryExists(file_path)) {
-        utility::filesystem::MakeDirectoryHierarchy(file_path);
+    // Resolve path.
+    const std::string resolved_data_root =
+            data_root.empty() ? LocateDataRoot() : data_root;
+    const std::string file_dir = resolved_data_root + "/" + prefix;
+    const std::string file_name =
+            utility::filesystem::GetFileNameWithoutDirectory(url);
```

Comment:
while in most cases, the URL's last part will contain the file name, it is not a compulsory requirement. 
For example: [Link](https://objects.githubusercontent.com/github-production-release-asset-2e65be/342474343/5e593811-31b1-452f-a226-7fb4cd62ac93?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20211225%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20211225T113824Z&X-Amz-Expires=300&X-Amz-Signature=50cd01725bf95a1d40069532731c92d867b880088de2078b7eb7ab7002e75096&X-Amz-SignedHeaders=host&actor_id=32535717&key_id=0&repo_id=342474343&response-content-disposition=attachment%3B%20filename%3Dtest_data_00.zip&response-content-type=application%2Foctet-stream)

In such cases, an optional `output_filename` would be helpful. We may skip this for now, and revisit here if we face any such case during the development of our dataset classes.

Reply:
Sounds good. My motivation is that the user shouldn't care about the saved file name, since the user will be using the dataset class, not the downloader. We can automatically generate a file name if the URL fails to resolve to a valid file name.